### PR TITLE
kcp-migration-broker and provisioner - change on helm hook annotations

### PR DESCRIPTION
**Pull request by talszabi**
**Description**
Moving kcp-migration-broker and kcp-migration-provisioner jobs to pre-install/pre-upgrade hook, as we observed issues that argocd did not execute this job during post-sync having conflict with long-running post-sync e2e test. 

**Related issue(s)**
[KCP Dev Provision failed: \"eu_access\" of relation \"gardener_config\" does not exist #3295](https://github.tools.sap/kyma/backlog/issues/3295)

### CODE REVIEW

**Review by ebensom — APPROVED** [review]

File: resources/kcp/templates/migrator-job.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ metadata:
         app: {{ .Chart.Name }}
         release: {{ .Release.Name }}
     annotations:
-        "helm.sh/hook": post-install,post-upgrade
+        "helm.sh/hook": pre-install,pre-upgrade
```
Comment (ebensom) [inline resources/kcp/templates/migrator-job.yaml:10]:
```suggestion
        "helm.sh/hook": post-install,pre-upgrade
```

File: resources/kcp/templates/migrator-job.yaml [hunk 2]
```diff
@@ -115,7 +115,7 @@ metadata:
...
```
Comment (ebensom) [inline resources/kcp/templates/migrator-job.yaml:118]:
```suggestion
        "helm.sh/hook": post-install,pre-upgrade
```

Comment (talszabi) [reply resources/kcp/templates/migrator-job.yaml]: resolved via: [bb66766](https://github.com/kyma-project/control-plane/commit/bb66766ab18f814e1a7d38af614bc97fffc5f338)

File: resources/kcp/templates/migrator-job.yaml (hunk 1 above)
Comment (talszabi) [reply resources/kcp/templates/migrator-job.yaml]: [bb66766](https://github.com/kyma-project/control-plane/commit/bb66766ab18f814e1a7d38af614bc97fffc5f338)

**Review by ebensom — APPROVED** [review]

**Review by life0215 — COMMENTED** [review]
Is this change mandatory, as we can set sync-wave for resources to be synced in sequence in post-sync

**Review by franpog859 — APPROVED** [review]

**Pull request merged**